logstash-core/spec/logstash/config/config_ast_spec.rb

```diff
@@ -158,6 +157,22 @@
 
         expect(compiled_number).to be == 8
       end
+
+      it "merges plugin options" do
+        parser = LogStashConfigParser.new
+        config = parser.parse(%q(
+          input {
+            example {
+              match => {'message' => '\d'}
+              match => {'message' => '\w'}
+            }
+          }
+        ))
+
+        compiled_hash = eval(config.recursive_select(LogStash::Config::AST::Hash).first.compile)
+
+        #expect(compiled_hash).to be == the_right_thing...
```

Comment:
I confirmed via manual testing that this change resolves the bug, but the above test does not exercise the affected code in `lscl.rb`.